### DISCUSSION TITLE
How to display linkage disequilibrium (R2) values as heatmaps

### DISCUSSION THREAD

**Discussion by apfuentes** [GMOD/jbrowse-components#4657] [Q&A]
Hi!

Thanks for developing JBrowse!

We wonder if it is possible to display linkage disequilibrium (R2) data as a heatmap, in a similar fashion as it is done for Hi-C data.

Could perhaps this be accomplished using [BEDPE](https://bedtools.readthedocs.io/en/latest/content/general-usage.html) or equivalent format files, since you could give the association value between two positions in the genome?

Thanks :)

**Comment by cmdcolin** [comment]
this would be great...I haven't done this in JBrowse 2 but I made a plugin for this in JBrowse 1 awhile back https://github.com/elsiklab/multivariantviewer

The way it worked actually required a server which would run the PLINK software on the user supplied VCF on the fly which would then return the LD matrix to the client side

it would be nice if there was a simple file format that contained the LD files that could be queried statically without a server side setup similar to the .hic format that we use for the Hi-C tracks, but could try to port over the jbrowse 1 plugin to jbrowes 2 as is also

**Reply by apfuentes** [reply]
Hi @cmdcolin, thank you for the prompt response!

The JBrowse1 plugin you mention sounds like a very clever and powerful solution, and I am sure it would be greatly appreciated if it becomes available in JBrowse2. However, the requirement of a server to compute LD on the fly could be resource costly for large genomic regions with lots of variants. 

Often for whole genome data, researchers subset VCF files (e.g., apply a filter to only retain SNPs every 50-100Kb) to get a reduced number of variants and compute LD using PLINK (or similar software) in their HPC resources. So, perhaps it would be easier to start from an already existing LD file. 

I wonder if there is a way to convert the LD file (e.g., computed with PLINK) into the HiC format (or similar) that receives a two loci pair and a value?

**Reply by cmdcolin** [reply]
re-purposing .hic (or pairix, another hic associated type https://github.com/4dn-dcic/pairix) could potentially work. the nice thing about it being .hic would be...we already have a hic track so it just automatically works :)

another file format that I found that could apply https://academic.oup.com/bioinformatics/article/39/2/btad092/7043094 ...

one thing that is perhaps relevant and which you can see in https://github.com/elsiklab/multivariantviewer/blob/master/img/ld.png is that it has lines connecting to the variant position, because the LDmatrix is not "mapped to bins of the genome" as hi-c is, but to specific variant positions. 

>However, the requirement of a server to compute LD on the fly could be resource costly for large genomic regions with lots of variants.


potentially that is true. i don't remember trying it on very large regions. it did have a "MAF" (minor allele frequency) filter that removed a lot of rare variants, which i think is somewhat common for LD calculations

**Reply by apfuentes** [reply]
Sorry for the delay in my reply, it's been a busy couple of weeks. Thanks for sharing these file formats! We're wrapping up end of the year activities, so I'll most likely try them out in January. I'll report back if I find anything interesting.

**Reply by cmdcolin** [reply]
hi @apfuentes happy new year. I was doing recent work looking at population vcf and thought linkage disequilibrium rendering would tie in. I stubbed out a plugin for it here https://github.com/cmdcolin/jbrowse-plugin-ldviewer

you can see some various brainstorm ideas in the readme, but interested if it could help
